components/add_users_to_team_modal/add_users_to_team_modal.tsx

```diff
@@ -202,11 +202,11 @@ export default class AddUsersToTeamModal extends React.PureComponent<Props, Stat
         let usersToDisplay: UserProfile[] = [];
         usersToDisplay = this.state.search ? this.state.searchResults : this.props.users;
         if (this.props.excludeUsers) {
-            const hasUser = (user: UserProfile) => !this.props.excludeUsers[user.id];
+            const hasUser = (user: UserProfile) => !this.props.excludeUsers.get(user.id);
             usersToDisplay = usersToDisplay.filter(hasUser);
         }
         if (this.props.includeUsers) {
-            const includeUsers = Object.values(this.props.includeUsers);
+            const includeUsers = this.props.includeUsers.values();
```

Comment:
In comparison with expanding an object, this works fine.

Nevertheless, the migration may give us lots of headaches, since "Object.values" may be "type-correct", and we won't see the problem until testing.

Reply:
That's very true. I've definitely ran into a few issues like that in the past when I've tried to use Maps elsewhere like on mobile. It's also really easy to write code like `map['foo'] = 'bar'` which is valid code even if it's not doing what you want. I think that's partly why I gave up trying to use Maps before we had TypeScript